sway-fmt/src/traversal_helper.rs

```diff
@@ -84,10 +85,67 @@ pub fn format_delineated_path(line: &str) -> String {
     line.chars().filter(|c| !c.is_whitespace()).collect()
 }
 
+/// Trims whitespaces and reorders compound import statements lexicographically
+/// a::{c, b, d::{self, f, e}} -> a::{b,c,d::{self,e,f}}
+fn sort_and_filter_use_expression(line: &str) -> String {
+    /// Tokenizes the line on separators keeping the separators.
+    fn tokenize(line: &str) -> Vec<String> {
+        let mut buffer: Vec<String> = Vec::new();
+        let mut current = 0;
+        for (index, separator) in line.match_indices(|c: char| c == ',' || c == '{' || c == '}') {
+            if index != current {
+                buffer.push(line[current..index].to_string());
+            }
+            buffer.push(separator.to_string());
+            current = index + separator.len();
+        }
+        if current < line.len() {
+            buffer.push(line[current..].to_string());
+        }
+        buffer
+    }
+    let tokens: Vec<String> = tokenize(line);
+    let mut buffer: Vec<String> = Vec::new();
+
+    fn sort_imports(tokens: &mut Iter<String>, buffer: &mut Vec<String>) {
+        let token = tokens.next();
+        match token.map(|t| t.trim()) {
+            None => return,
+            Some(",") => (),
+            Some("{") => {
+                let mut inner_buffer: Vec<String> = Vec::new();
+                sort_imports(tokens, &mut inner_buffer);
+                if let Some(buff) = buffer.last_mut() {
+                    buff.push_str(inner_buffer[0].as_str());
```

Comment:
Is it a guarantee/invariant that `inner_buffer[0]` is always present, i.e. the buffer is never empty after `sort_imports`?

Reply:
updated for the case where it can be empty - when there is malformed data with an open '{' but no corresponding closing '}'